wallet/wallet.go

```diff
@@ -4195,6 +4196,91 @@ func (w *Wallet) ChainParams() *chaincfg.Params {
 	return w.chainParams
 }
 
+// scriptChangeSource is a ChangeSource which is used to
+// receive all correlated previous input value.
+type scriptChangeSource struct {
+	version uint16
+	script  []byte
+}
+
+func (src *scriptChangeSource) Script() ([]byte, uint16, error) {
+	return src.script, src.version, nil
+}
+
+func (src *scriptChangeSource) ScriptSize() int {
+	return len(src.script)
+}
+
+func makeScriptChangeSource(address string, version uint16, params *chaincfg.Params) (*scriptChangeSource, error) {
+	destinationAddress, err := dcrutil.DecodeAddress(address, params)
+	if err != nil {
+		return nil, err
+	}
+
+	var script []byte
+	if addr, ok := destinationAddress.(V0Scripter); ok && version == 0 {
+		script = addr.ScriptV0()
+	} else {
+		script, err = txscript.PayToAddrScript(destinationAddress)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	source := &scriptChangeSource{
+		version: version,
+		script:  script,
+	}
+
+	return source, nil
+}
+
+// FundRawTransaction adds inputs sourced from the wallet matching the total
+// output value to a transaction.
+func (w *Wallet) FundRawTransaction(ctx context.Context, outputs []*wire.TxOut, account uint32,
+	changeAddr *string, feeRate *float64, targetConf *int32) (*types.FundRawTransactionResult, error) {
+	var changeSource txauthor.ChangeSource
+	var err error
+	if changeSource != nil {
+		changeSource, err = makeScriptChangeSource(*changeAddr, 0, w.chainParams)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	minConf := int32(1)
+	if targetConf != nil {
+		minConf = *targetConf
+	}
+
+	relayFee := w.RelayFee()
+	if feeRate != nil {
+		relayFee, err = dcrutil.NewAmount(*feeRate)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	aTx, err := w.NewUnsignedTransaction(outputs, relayFee, account, minConf,
+		OutputSelectionAlgorithmDefault, changeSource)
+	if err != nil {
+		return nil, err
+	}
+
+	txBytes, err := aTx.Tx.Bytes()
+	if err != nil {
+		return nil, err
+	}
+
+	feeEst := txrules.FeeForSerializeSize(relayFee, aTx.EstimatedSignedSerializeSize)
```

Comment:
this is the wrong fee... Because the transaction is now funded, the fee can be calculated as an absolute value (total input value - total output value) and there's no reason to estimate it.  This should also be performed by the rpc handler on a returned wire.MsgTx.